ds

```diff
@@ -203,13 +257,55 @@ _save() {
     [[ ! -e $_OUTPUT_DIR ]] && mkdir "$_OUTPUT_DIR"
     save
 }
-for arg in "$@"; do
+
+_EXCLUDE_OVERRIDES=()
+_EXCLUDES=()
+_ACTION=""
+_DOMAIN=""
+
+while [[ $# -gt 0 ]]; do
```

Comment:
This loop becomes infinite loop if `ds` is executed without any command

Reply:
Sorry for not being able to respond for a while. Been busy and will still likely be so for a while... :(
Anyways, did a quick test and requested a change.

Reply:
No worries! When I run `./ds`, I get its usage message and an exit, not an infinite loop. Is there a way for me to reproduce this infinite loop?

Reply:
Oops, I misexplained it, my bad.

When `ds` is launched with any invalid arguments that is not handled by switch `case` inside `while` loop, infinite loop happens because `shift` never gets called which means `$#` never decrements.

Reply:
Oh, thank you! I've added a test and fix.